addons/mass_mailing/wizard/mail_compose_message.py

```diff
@@ -66,4 +66,12 @@ def get_mail_values(self, res_ids):
                     'notification': mass_mailing.reply_to_mode == 'thread',
                     'auto_delete': not mass_mailing.keep_archives,
                 })
+                if not (res[res_id].get('email_to') or res[res_id].get('recipient_ids')):
```

Comment:
You should instead put the mail in state cancel like done before for blacklist and stuff like that. Avoid modifying res, and put correct values in it instead.